elftools/construct/core.py

```diff
@@ -493,23 +493,17 @@ def __init__(self, mincount, maxcout, subcon):
         self._set_flag(self.FLAG_DYNAMIC)
     def _parse(self, stream, context):
         obj = ListContainer()
-        c = 0
-        try:
-            if self.subcon.conflags & self.FLAG_COPY_CONTEXT:
-                while c < self.maxcout:
-                    pos = stream.tell()
-                    obj.append(self.subcon._parse(stream, context.__copy__()))
-                    c += 1
-            else:
-                while c < self.maxcout:
-                    pos = stream.tell()
-                    obj.append(self.subcon._parse(stream, context))
-                    c += 1
-        except ConstructError as ex:
-            if c < self.mincount:
-                raise RangeError("expected %d to %d, found %d" %
-                    (self.mincount, self.maxcout, c), ex)
-            stream.seek(pos)
+        ctx = context.__copy__ if self.subcon.conflags & self.FLAG_COPY_CONTEXT else (lambda: context)
```

Comment:
If at all possible, I would prefer to minimize changes inside `elftools/construct`

Is there a simpler way to make the type checker happy? Just predeclaring pos?

Reply:
Minimum would be an additional `pos = -1` before the `try`, e.g.
```diff
--- a/elftools/construct/core.py
+++ b/elftools/construct/core.py
@@ -496,1 +496,1 @@ class Range(Subconstruct):
-        c = 0
+        c = pos = -1
```

Reply:
Now also in #609